netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -286,6 +280,13 @@ public void run() {
       };
     }
     channel = regFuture.channel();
+    // For non-epoll based channel, the option will be ignored.
+    if (keepAliveTimeNanos != KEEPALIVE_TIME_NANOS_DISABLED && channel instanceof AbstractEpollStreamChannel) {
```

Comment:
Instead of adding an epoll dependency, let's continue as was already done and use reflection. `Class.forName("io.netty.channel.epoll.AbstractEpollStreamChannel").isInstance(channel)` (and handle errors)